src/dso_api/dynamic_api/routers.py

```diff
@@ -26,6 +26,19 @@
     from schematools.contrib.django.models import DynamicModel
 
 
+def _environment_enabled_datasets(enabled_datasets):
+    if (
+        settings.DATAPUNT_ENVIRONMENT == "production"
+        and "acceptance_only_datasets" in settings.AMSTERDAM_SCHEMA
+    ):
+        enabled_datasets = filter(
+            lambda ds: ds.schema.id
+            not in settings.AMSTERDAM_SCHEMA["acceptance_only_datasets"],
+            enabled_datasets,
+        )
+    return enabled_datasets
```

Comment:
```suggestion
        return [
            ds for ds in enabled_datasets
            if not in settings.AMSTERDAM_SCHEMA["acceptance_only_datasets"]
        ]
```

ik zie de meerwaarde van de (oude) filter functie hier niet zo, plus dat het geen lijst teruggeeft maar een iterator die je maar eenmalig kan uitlezen.

Reply:
Er wordt gefilterd op of ds.schema.id in settings.AMSTERDAM_SCHEMA["acceptance_only_datasets"] zit. Niet ds zelf